### ISSUE TITLE
Change FFI files to use ES6 modules rather than CommonJS

### ISSUE THREAD

**Issue by Pauan** [purescript/purescript#2575]
This is a backwards-incompatible change, but I think we will need to do it eventually.

It also depends upon `rollup-plugin-purs` replacing `psc-bundle`, so we can't do it right now anyways.

**Comment by kevinbarabash** [comment]
I've been experimenting with generating codegen of ES6 module.  For it to work I also need to upgrade the foreign.js files to be ES6 modules as well.  I did this by hand in test app I'm working on, but I believe that the process can be automated.  The one thing that's tricky is that ES6 `export` statements don't allow special characters in the exports so converting `exports["foo'"] = function () { ... }` to `export const foo' = function () { ... }` won't work.  Instead, I converted it to `export const foo$prime = function () { ... }`.

This requires changes to the compiler to which would converts the foreign import to `$foreign["foo'"]` to instead output `$foreign["foo$prime"]`.  I think treating JS keywords by keeping them in `$$keyword` form would simplify the necessary changes.
